graph.go

```diff
@@ -15,12 +16,20 @@ func (g *graph) FormatStateMachine(sm *StateMachine) string {
 	var sb strings.Builder
 	sb.WriteString("digraph {\n\tcompound=true;\n\tnode [shape=Mrecord];\n\trankdir=\"LR\";\n\n")
 
-	for _, sr := range sm.stateConfig {
+	var stateList []*stateRepresentation
```

Comment:
Preallocate the slice with the size of sm.stateConfig

Reply:
Done. Did the same for `sr.TriggerBehaviours` but there it will still grow if there's more than one behaviour for the same trigger.